demo/src/main/java/com/cardinalblue/kraftshade/demo/ui/screen/media3/TransformerWithKraftShaderPipeline.kt

```diff
@@ -0,0 +1,200 @@
+import com.cardinalblue.kraftshade.shader.builtin.SaturationKraftShader
+import kotlin.math.abs
+
+@OptIn(UnstableApi::class)
+@Composable
+fun TransformerWithKraftShaderPipeline() {
+    NetworkAwareScreen(
+        additionalDescription = "We need network to stream the sample video for this demo"
+    ) {
+        // You can either use MediaMetadataRetriever to get the aspect ratio or add listener to listen
+        // for the video frame sizes. However, Player.Listener won't be working when you use
+        // setVideoEffects on ExoPlayer. This is an existing issue that might not be resolved soon.
+        // See https://github.com/androidx/media/issues/2505
+        // The black bounds are due to callback not being called.
+        var videoAspectRatio by remember { mutableFloatStateOf(16f / 9) }
+
+        val context = LocalContext.current
+        var exportedUri by remember { mutableStateOf<String?>(null) }
+        val exoPlayer = remember {
+            ExoPlayer.Builder(context)
+                .build()
+                .apply { playWhenReady = true }
+        }
+        LaunchedEffect(exportedUri) {
+            if (exportedUri == null) return@LaunchedEffect
+            exoPlayer.setMediaItems(
+                listOf(
+                    MediaItem.Builder()
+                        .setUri(exportedUri!!)
+                        .build()
+                )
+            )
+            exoPlayer.playWhenReady = true
+            exoPlayer.prepare()
+        }
+        Column(
+            modifier = Modifier
+                .fillMaxSize()
+                .padding(16.dp),
+            horizontalAlignment = Alignment.CenterHorizontally,
+            verticalArrangement = spacedBy(16.dp)
+        ) {
+            AndroidView(
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .aspectRatio(videoAspectRatio),
+                factory = {
+                    PlayerView(context).apply {
+                        player = exoPlayer
+                    }
+                }
+            )
+
+            DisposableEffect(Unit) {
+                onDispose {
+                    exoPlayer.release()
+                }
+            }
+
+            var from by remember { mutableIntStateOf(0) }
+            var to by remember { mutableIntStateOf(20) }
+            TextField(
+                value = from.toString(),
+                onValueChange = { from = it.toIntOrNull() ?: 0 },
+                label = { Text("From (s)") },
+                keyboardOptions = KeyboardOptions.Default.copy(keyboardType = KeyboardType.Number)
+            )
+            TextField(
+                value = to.toString(),
+                onValueChange = { to = it.toIntOrNull() ?: 20 },
+                label = { Text("To (s)") },
+                keyboardOptions = KeyboardOptions.Default.copy(keyboardType = KeyboardType.Number)
+            )
+
+
+            var exporting by remember { mutableStateOf(false) }
```

Comment:
Should we just make this a mutableSharedFlow and then collect from it to start the processing? And we can still have a state of "exporting" for the processing status.